src/serverenvironment.cpp

```diff
@@ -1726,8 +1731,13 @@ void ServerEnvironment::getAddedActiveObjects(PlayerSAO *playersao, s16 radius,
 	if (player_radius_f < 0.0f)
 		player_radius_f = 0.0f;
 
-	m_ao_manager.getAddedActiveObjectsAroundPos(playersao->getBasePosition(), radius_f,
-		player_radius_f, current_objects, added_objects);
+	if (!playersao->isEffectivelyObservedBy(playersao->getPlayer()->getName()))
+		throw ModError("Player does not observe itself");
```

Comment:
Observation, no action needed. This can be triggered by attaching to an object that is not observed, or when removing the player from the parent's observer list.
```Lua
minetest.register_chatcommand("p", {
	func = function(name, param)
		local player = minetest.get_player_by_name(name)
		local pos = player:get_pos()

		local obj = core.spawn_item(pos, "default:dirt")
		obj:set_observers({["foo"] = true, ["bar"] = true })

		player:set_attach(obj)
		return true, "OK!"
	end
})
```

It is similar to the situation in the previous review: https://github.com/minetest/minetest/pull/13987#discussion_r1677192447 but very difficult to make this less error-prone. The main issue is that this error cannot be captured by `pcall` (there is no active callback).